filip/clients/ngsi_v2/cb.py

```diff
@@ -598,25 +601,29 @@ def delete_entity(self,
             raise
 
         if delete_devices:
-            from filip.clients.ngsi_v2 import IoTAClient
-            if iota_client:
-                iota_client_local = deepcopy(iota_client)
-            else:
-                warnings.warn("No IoTA-Client object provided! "
-                              "Will try to generate one. "
-                              "This usage is not recommended.")
+            if entity_type:
+                from filip.clients.ngsi_v2 import IoTAClient
+                if iota_client:
+                    iota_client_local = deepcopy(iota_client)
+                else:
+                    warnings.warn("No IoTA-Client object provided! "
+                                  "Will try to generate one. "
+                                  "This usage is not recommended.")
 
-                iota_client_local = IoTAClient(
-                    url=iota_url,
-                    fiware_header=self.fiware_headers,
-                    headers=self.headers)
+                    iota_client_local = IoTAClient(
+                        url=iota_url,
+                        fiware_header=self.fiware_headers,
+                        headers=self.headers)
 
-            for device in iota_client_local.get_device_list(
-                    entity_names=[entity_id]):
-                if device.entity_type == entity_type:
-                    iota_client_local.delete_device(device_id=device.device_id)
+                for device in iota_client_local.get_device_list(
+                        entity_names=[entity_id]):
+                    if device.entity_type == entity_type:
+                        iota_client_local.delete_device(device_id=device.device_id)
 
-            iota_client_local.close()
+                iota_client_local.close()
+            else:
```

Comment:
Why should it not be possible to delete a device without providing entity_type?

Reply:
I think the idea is that only the combination of `entity_name` and `entity_type` can uniquely identify an entity. But this is a decision we can choose

Reply:
It will be revised to delete all devices matched with that "id"